worker/uniter/relation/relationer.go

```diff
@@ -105,7 +110,15 @@ func (r *Relationer) CommitHook(hi hook.Info) error {
 		panic("implicit relations must not run hooks")
 	}
 	if hi.Kind == hooks.RelationBroken {
-		return r.die()
+		err := r.die()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		rel := r.ru.Relation()
+		if r.suspended {
+			return rel.SetStatus(relation.Suspended)
```

Comment:
Oof, I just realised something: I don't think we can set relation status from any old unit. I think it has to be the application leader, otherwise they could trample each other.

Reply:
done